controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -57,10 +76,11 @@
      */
     public void createStream(String scope, String streamName, int minNumSegments, Duration latency) {
         DYNAMIC_LOGGER.incCounterValue(CREATE_STREAM, 1);
-        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(OPEN_TRANSACTIONS, scope, streamName), 0);
+        // We need to initialize the number of initial segments, but prevent this process to continue reporting the
+        // segment count for this Stream. This will avoid multiple controllers to report conflicting values.
         DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName), minNumSegments);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, streamName), 0);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_MERGES, scope, streamName), 0);
+        executor.schedule(() -> DYNAMIC_LOGGER.freezeGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName)),
+                statsOutputFrequencySeconds + 1, TimeUnit.SECONDS);
```

Comment:
Why "+1" ?
Also, what does freeGaugeValue do?

Reply:
`freezeGaugeValue` removes the metric from the metrics cache, so it is not reported anymore. By delaying the execution `freezeGaugeValue` one reporting period, we ensure that we report the initial number of segments for this Stream and then the Controller will stop reporting it. Regarding the `+ 1`, it was just to be sure that the deletion occurs at least after a full `statsOutputFrequencySeconds` period completes, but maybe I can delete it for clarity.